Packages/Tracking Preview/Examples~/HandRays/Scripts/HandRayRenderer.cs

```diff
@@ -0,0 +1,162 @@
+using Leap.Unity;
+using Leap.Unity.Interaction;
+using Leap.Unity.Interaction.PhysicsHands;
+using Leap.Unity.Preview.HandRays;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public abstract class HandRayRenderer : MonoBehaviour
+{
+    [SerializeField] private HandRay _handRay;
+
+    [Header("Rendering")]
+    [SerializeField] private LineRenderer _lineRenderer;
+
+    [Header("Trail")]
+    [SerializeField] private Transform _trailRendererTransform;
+    [SerializeField] private float _trailRendererYOffset = 0.04f;
+    private TrailRenderer _trailRenderer;
+
+    [Tooltip("Note that the Interaction Engine and Physics Hands layers will be ignored automatically.")]
+    [Header("Layer Logic")]
+    [SerializeField] private List<SingleLayer> _layersToIgnore = new List<SingleLayer>();
+
+    protected LayerMask _layerMask;
+
+    [Header("Events")]
+    public UnityEvent<RaycastHit> OnRayUpdate;
+
+    private void Start()
+    {
+        _layerMask = -1;
+        InteractionManager interactionManager = FindObjectOfType<InteractionManager>();
+        if (interactionManager != null)
+        {
+            _layerMask ^= interactionManager.contactBoneLayer.layerMask;
+        }
```

Comment:
It feels like Handrays being dependent on IE is the wrong way around?

Can we make IE request its layers be added instead?

Reply:
Actually, this would cause tracking to have a dependency on preview